src/apps/loyalty/containers/login/__tests__/index.tsx

```diff
@@ -1,33 +1,98 @@
+// 3. And finally we reset the mock after each test to ensure our mocked implementation doesn’t bleed into other tests.
+jest.mock("isomorphic-fetch")
 import Login from "../index"
 
 describe("login", () => {
+  const formConfig = {
+    baseUrl: "/loyalty",
+    url: "/log_in",
+    csrfToken: "csrf",
+    facebookPath: "/facebook",
+    twitterPath: "/twitter",
+  }
+
+  afterEach(() => {
+    require("isomorphic-fetch").mockReset()
+  })
+
   it("renders the snapshot", () => {
-    const formConfig = {
-      url: "/log_in",
-      csrfToken: "csrf",
-      facebookPath: "/facebook",
-      twitterPath: "/twitter",
-    }
     const login = renderer.create(<Login form={formConfig} />)
     expect(login).toMatchSnapshot()
   })
 
   it("renders the social buttons with the correct hrefs", () => {
-    const formConfig = {
-      url: "/log_in",
-      csrfToken: "csrf",
-      facebookPath: "/facebook",
-      twitterPath: "/twitter",
-    }
-    const login = ReactTestUtils.renderIntoDocument(<Login form={formConfig} />) as Login
-    const aTags = ReactTestUtils.scryRenderedDOMComponentsWithTag(login, "a") as HTMLAnchorElement[]
+    const login = TestUtils.renderIntoDocument(<Login form={formConfig} />) as Login
+    const aTags = TestUtils.scryRenderedDOMComponentsWithTag(login, "a") as HTMLAnchorElement[]
     const facebookLink = aTags.find(tag => tag.href === "/facebook")
     expect(facebookLink).toBeTruthy()
     const twitterLink = aTags.find(tag => tag.href === "/twitter")
     expect(twitterLink).toBeTruthy()
   })
+
+  it("displays correct error message", (done) => {
+    const fetch = require("isomorphic-fetch")
+    fetch.mockImplementation(() => {
+      return Promise.resolve({
+        url: "http://localhost:3000/log_in?error=Invalid%20email%20or%20password",
+        status: 404,
+      })
+    })
+
+    const login = TestUtils.renderIntoDocument(<Login form={formConfig} />) as Login
+
+    TestUtils.Simulate.submit(TestUtils.findRenderedDOMComponentWithTag(login, "form"))
+
+    // Wait till the next event loop tick to assert on the expected DOM state,
+    // because `setState` isn’t going to re-render until the next tick either.
+    setImmediate(() => {
+      const error = TestUtils.findRenderedDOMComponentWithClass(login, "error")
+      expect(error.textContent).toBe("Invalid email or password")
+      done()
+    })
+  })
+
+  it("successful redirects after login", () => {
+    const fetch = require("isomorphic-fetch")
+    fetch.mockImplementation(() => {
+      return Promise.resolve({
+        status: 200,
+      })
+    })
+
+    const mockRedirectTo = Login.prototype.redirectTo = (url: string) => {
+      expect(url).toEqual("/loyalty/inquiries")
+    }
+
+    const login = <Login form={formConfig} />
+    const renderedComponent = TestUtils.renderIntoDocument(login) as Login
+
+    TestUtils.Simulate.submit(TestUtils.findRenderedDOMComponentWithTag(renderedComponent, "form"))
```

Comment:
Are you sure this test works as expected? Seeing as this test does not work async like the others, it is possible that your expectation on line 69 simply never runs, thus is a false positive.